persist/adapter.go

```diff
@@ -72,3 +73,21 @@ type Adapter interface {
 	// This is part of the Auto-Save feature.
 	RemoveFilteredPolicy(sec string, ptype string, fieldIndex int, fieldValues ...string) error
 }
+
+// AdapterWithContext provides a context-aware interface for Casbin adapters.
+type AdapterWithContext interface {
```

Comment:
Rename to `ContextAdapter`, put to `adapter_context.go`

Like: https://github.com/casbin/casbin/blob/master/persist/adapter_filtered.go